custom_components/tahoma/__init__.py

```diff
@@ -111,47 +111,55 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         scenarios = await client.get_scenarios()
         gateways = await client.get_gateways()
     except BadCredentialsException:
-        _LOGGER.error("invalid_auth")
+        _LOGGER.error("Invalid authentication.")
         return False
     except TooManyRequestsException as exception:
-        _LOGGER.error("too_many_requests")
+        _LOGGER.error("Too many requests, try again later.")
         raise ConfigEntryNotReady from exception
     except (TimeoutError, ClientError, ServerDisconnectedError) as exception:
-        _LOGGER.error("cannot_connect")
+        _LOGGER.error("Failed to connect.")
         raise ConfigEntryNotReady from exception
     except MaintenanceException as exception:
-        _LOGGER.error("server_in_maintenance")
+        _LOGGER.error("Server is down for maintenance.")
         raise ConfigEntryNotReady from exception
     except Exception as exception:  # pylint: disable=broad-except
         _LOGGER.exception(exception)
         return False
 
-    update_interval = entry.options.get(CONF_UPDATE_INTERVAL, DEFAULT_UPDATE_INTERVAL)
+    update_interval = timedelta(
+        seconds=entry.options.get(CONF_UPDATE_INTERVAL, DEFAULT_UPDATE_INTERVAL)
+    )
```

Comment:
Why change this?

Reply:
I had to move the following debug from our __init__ method in the Data Update Coordinator. Moving up the timedelta made it easier to reuse this variable for the DataUpdateCoordinator and debug logging.

```python
_LOGGER.debug(
        "Initialized DataUpdateCoordinator with %s interval.", str(update_interval)
    )
```